lib/perf_hooks.js

```diff
@@ -608,13 +610,20 @@ function monitorEventLoopDelay(options = {}) {
     const errors = lazyErrors();
     throw new errors.ERR_INVALID_OPT_VALUE.RangeError('resolution', resolution);
   }
-  return new ELDHistogram(new _ELDHistogram(resolution));
+  return new Histogram(new ELDHistogram(resolution));
+}
+
+function monitorRequestWrapLatency() {
+  if (!rwlHistogram)
```

Comment:
Would it be possible to allow several histograms to be filled in parallel? I think the limitation to a single instance here could easily result in conflicts between various monitoring tools like APM, launchers,... reporting this continuously to their parents and dedicated use by a developers during testing.

Reply:
Doing do would have a significant impact on overall performance, with multiple monitors in place, multiple branches, and multiple histograms to update on each ReqWrap completion. I'd rather not go down that path if we can avoid it.

Reply:
@jasnell if the implementation would only track raw data and add markers in case the functionality is called again, then it would be possible to calculate each call individually without having to track things again. So performance wise it should be possible, no?

Reply:
I would not expect "unnumbered" users of this API and updating a handfull histograms should be not that expensive compared to the actual work done within an uv operation (just a guess, not measured).
Maybe use something cheaper then the histogram with just min/max/sum/cnt?

Even I'm not a fan of this but we could also think about `monitorRequestWrapLatency()` returning `out of resources` or similar if the number gets too high.

If we go for a single user API we should clearly document this limitation as it is different from `monitorEventLoopDelay()` and the WIP `monitorHeapStatistics()`.

Reply:
I really believe that needlessly over-complicates things here. Before going down that road, I'd prefer a bit more justification that the current implementation would be problematic.

Reply:
Thinking about it a bit further, I think there's a way I can a least limit the impact in the common case and still allow the multiple histograms. I'll make another pass at it.